src/auth.rs

```diff
@@ -53,28 +51,26 @@ pub struct Authorization {
 /// request authentication, and for authenticating outgoing client requests.
 #[derive(Clone, Debug, PartialEq)]
 pub enum AuthData {
-    /// HTTP Basic auth.
-    Basic(Basic),
-    /// HTTP Bearer auth, used for OAuth2.
-    Bearer(Bearer),
+    /// HTTP Basic auth - username and password.
+    Basic(String, String),
+    /// HTTP Bearer auth, used for OAuth2 - token.
+    Bearer(String),
     /// Header-based or query parameter-based API key auth.
     ApiKey(String),
 }
 
 impl AuthData {
     /// Set Basic authentication
     pub fn basic(username: &str, password: &str) -> Self {
-        AuthData::Basic(Basic {
-            username: username.to_owned(),
-            password: Some(password.to_owned()),
-        })
+        let basic = Header::basic(username, password);
+        AuthData::Basic(basic.username().to_owned(), basic.password().to_owned())
```

Comment:
I've just realised this is a pretty pointless implementation - I'm cnnstucting the header for no reason.   I feel like ideally, we would return https://docs.rs/headers/latest/headers/authorization/struct.Basic.html here - but that has no constructor beyond a decode from a `HeaderValue`.  (Same applies to the Bearer variant).  We could make the inner type an authorization header but that feels wrong too.

Thoughts welcome!